src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BackgroundAspectEffect/BackgroundAspectEffectRouter.ios.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using UIKit;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+using Effects = Xamarin.CommunityToolkit.iOS.Effects;
+
+[assembly: ExportEffect(typeof(Effects.BackgroundAspectEffectRouter), nameof(BackgroundAspectEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.iOS.Effects
+{
+	public class BackgroundAspectEffectRouter : PlatformEffect
+	{
+		UIImageView? imageView;
+
+		CancellationTokenSource? cancellationTokenSource;
+
+		public BackgroundAspectEffectRouter()
+		{
+			cancellationTokenSource = new CancellationTokenSource();
+		}
+
+		protected override void OnAttached()
+			=> Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource!.Token);
+
+		protected override void OnDetached()
+			=> ClearBackgroundImage();
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (!args.PropertyName.Equals(BackgroundAspectEffect.AspectProperty.PropertyName))
+				return;
+
+			cancellationTokenSource?.Cancel();
+
+			Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource!.Token);
+		}
+
+		async Task ApplyBackgroundImageAsync()
+		{
+			var contentPage = Element as ContentPage;
+
+			if (Element == null || contentPage == null || contentPage.BackgroundImageSource == null)
+				return;
+
+			var image = await GetBackgroundImageFromSource(contentPage.BackgroundImageSource);
+
+			if (image is null)
```

Comment:
we don't use pattern matching for null check
```suggestion
			if (image == null)
```